File: euler-number/python/euler.py
"""
Heru Handika
16 September 2020
Euler's number implementation on python
Using more verbose while loop.
"""

def calculate_factorial(n):
    if n == 0 or n == 1:
        return 1
    else:
        return n * calculate_factorial(n-1)

def calculate_euler_number(n):
    counts = 1
    total = 0.0
    while counts < n:
        total += 1.0 / calculate_factorial(counts)
        counts += 1
        
    return total + 1

def main():
    n = 10
    euler_number = calculate_euler_number(n)

    print(f"Euler's number is {euler_number}")

if __name__ == "__main__":
    main()

File: euler-number/julia/euler.jl
#=
Heru Handika
18 September 2020
Euler's number implementation in julia
=#

using Printf

function calculate_factorial(n)
    if n == 0 || n == 1
        return 1
    else
        return n * calculate_factorial(n - 1)
    end
end

function main()
    euler_number::Float64 = 1.0
    for i = 1:10
        euler_number = euler_number + 1.0 / calculate_factorial(i)
    end
    @printf("Euler number: %.8f\n", euler_number)

end

if abspath(PROGRAM_FILE) == @__FILE__
    main()
end